draft/draft-ietf-bfd-optimizing-authentication.xml

```diff
@@ -392,12 +395,51 @@
         excepting that Auth Type is still TBD and that Reserved is set to 2.
       </t>
 
-      <section title="Error Handling">
+      <section title="Receipt Using Optimized Authentication">
+        <t>
+          The procedures for authenticating BFD Control packets is similar to
+          the existing procedures covered in <xref target="RFC5880" section="6.7"/>. 
+          The Optimized Authentication mode provides requirements for
+          authentication key processing and how the result of the authentication
+          operation is stored in the authentication section for that mode.
+          Similarly, the value of the Auth Len field is specified by the
+          Optimized Authentication mode in each of their specifications.
+        </t>
+        <t>
+          The following common procedures apply to authenticating BFD Control
+          packets utilizing Optimized Authentication:
+        </t>
+        <t>
+          If the received BFD Control packet does not contain an Authentication
+          Section, or the Auth Type is not a supported Optimized Authentication
+          Auth Type, then the received packet MUST be discarded.
+        </t>
         <t>
 	  If the received BFD Control packet contains an optimized
 	  authentication type using these procedures and the Optimized
           Authentication Mode field is not 1 or 2, then the received packet
           MUST be discarded.
+        </t>
+        <t>
+          If the Auth Len field is not equal to a value appropriate for the
+          Optimized Authentication Mode field, the packet MUST be discarded.
+        </t>
+        <t>
+          If bfd.AuthSeqKnown is 1, examine the Sequence Number field.  If the
+          sequence number lies outside of the range of bfd.RcvAuthSeq+1 to
+          bfd.RcvAuthSeq+(3*Detect Mult) inclusive (when treated as an unsigned
+          32-bit circular number space, the received packet MUST be discarded.
+        </t>
+        <t>
+          Otherwise (bfd.AuthSeqKnown is 0), bfd.AuthSeqKnown MUST be set to 1,
+          bfd.RcvAuthSeq MUST be set to the value of the received Sequence
+          Number field, and the received packet MUST be accepted.
```

Comment:
The word Otherwise to me implies the opposite of what the previous paragraph is talking about, i.e., it is talking about the case when bfd.AuthSeqKnown == 0, so maybe what is inside the parenthesis is redundant. Though it appears that Otherwise it is referring to the case when the sequence number falls inside the range of bfd.RcvAuthSeq+1 and bfd.RcvAuthSeq+3. The sentence structure or the way the paragraphs are laid out seems confusing to me. Would it help to say:

"Otherwise, if the sequence number falls within the range of bfd.RcvAuthSeq+1 and bfd.RcvAuthSeq+(3*Detect Mult) inclusive, and bfd.AuthSeqKnown is set to 1, the received packet MUST be accepted."?

Reply:
The Otherwise section is straight from 5880.  Is it just because of how it looks in this text here, or do you similarly find the 5880 text to have the same considerations?

The otherwise is talking about the case where the sequence number is NOT known.  In other words, we're bootstrapping sequence numbers for a fresh session.